### ISSUE TITLE
[Bug]: website run by sail does not even after I quit Herd

### ISSUE THREAD

**Issue by ataldev** [beyondcode/herd-community#288]
### Operating system version

macOS Ventura 13

### System architecture

Intel (x86)

### Herd Version

1.3.1

### Bug description

When Herd is open, all the added directories to Herd open with .test at the end. I also have some apps running via Laravel Sail. When Herd is open, the apps run via Sail do not open with .test at the end. Even after I quit Herd entirely, they don't open I get this: ERR_NAME_NOT_RESOLVED
To solve this, I always have to restart my computer.

### Steps to reproduce

Run a project with ```sail up -d``` then open Herd, close Herd again, and try to access the app run with Sail.

**Comment by sschlein** [comment]
Does this also happen if you stop all Herd services instead of just quitting the app?

Running Sail and Herd in parallel is generally not supported because they use identical ports, so you have to sail down before you start your Herd services and vice versa. 

Does this help?

**Comment by ataldev** [comment]
I tried that as well. Didn't work. 🙁

**Comment by mpociot** [comment]
Are all services marked as running? Name not resolved means that DNSMasq is not propertly setup/running.

**Comment by mpociot** [comment]
Did you find some time to look into this @ataldev ?

**Comment by cs02lk1** [comment]
Hi @mpociot ,

I am facing the same issue. I have been working with Laravel Sail for some time and setup working with .test domains. After installing Laravel Herd,  .test domains are not reachable unless laravel herd is running. Also note that each sail project is set to run on a specific port instead of the default 80 for http.  The not working .test domains applies to even projects which I have not yet parked in Herd.

As soon as I start Laravel Herd, .test domain is reachable.

If I Stop All services from herd menu, .test domain is not reachable again. They only become available when Herd and herd services are running.

**Comment by sschlein** [comment]
@cs02lk1 If you close Herd or stop all services from the Herd menu, this stops the DNS resolver and this is why .test domains don't work. You cannot run Herd and Sail apps in parallel because only one app can spin up a DNS service at the same time.

**Comment by cs02lk1** [comment]
@sschlein 

Thanks for the reply.

I understand now that the problem seems to be with the DNS resolver and when trying to run in parallel, but even when I stop everything (Stop Herd Services, Quit Herd, Quite Docker) and just startup docker and run "sail up" for one of the projects, it is still not resolved. Shouldn't that work?

**Comment by sschlein** [comment]
Can you check if there's still a dnsmasq process running on your machine?

Maybe the process isn't terminated correctly after you quit Herd and it blocks the port so that Sail can't spin up its dns.

**Comment by cs02lk1** [comment]
Yeah I did

- ps -ef | grep dnsmasq shows Herd dnsmasq services running
- Stop All Services from Herd
- Exit/Quit Laravel Herd
- ps -ef | grep dnsmasq shows nothing running
- exit iterm

- Exit/Quit Docker
- Run Docker
- Run New Terminal Window
- make sure no dnsmasq services are running
- navigate to project folder and run "sail up -d"

Site not available (ERR_NAME_NOT_RESOLVED), pinging the .test domain not responding as well (Unknown host)

**Comment by cs02lk1** [comment]
I used to work with only Laravel Sail and everything worked. You are correct that the default url to access a laravel sail app is http:localhost, however it can be setup to work on the .test domain as well using the service name. This was already setup and working on my local system.

I wanted to check Laravel Herd as well and as soon as I installed it, the discussed problem has been happening since.  The projects (even the ones which I have not included in the folders list of Herd) are not accessible unless Herd is running. 

As a workaround I can change my workflow and use a different TLD domain for the projects when I use Sail (i.e. .example or .local). I already tried this and it seems to work. I just need to update /etc/hosts file accordingly to direct the domain to 127.0.0.1 for each project domain. 

A few things I noticed
- If I edit /etc/hosts and add hrinnovate.test (the project domain) then it works fine and the project is accessible through the browser through the browser (and Laravel Herd, dnsmasq not running). 
- Is the file /etc/resolv.conf created by Laravel Herd during instalation? As I am not sure, I haven't investigate this before the issue occured. If yes, could that be picked up by MacOs Dns service (instead of Herd's dnsmasq) somehow, so we have the conflict with the docket dns?

What if Uninstall Laravel Herd. Will all files generated (apart from it's local folder) be removed or do I need to manually remove them ? Would that be safe and would it be the case where my system returns to the previous state where everything seem to work? :)

**Comment by cs02lk1** [comment]
@mpociot 

You are right. Thanks for pitching in. I was close to figuring out that the file /etc/resolver/test was the config being picked up by the macOS DNS which seems to take priority over docker DNS. 

My development workflow greatly benefits from Herd's awesome features, but also I need to be able to use Sail for specific things. 

I will go ahead with the option of using a different TLD domain for the laravel sail projects giving me the best of two worlds with minimal disruption and effort.

One last educational question :)
- I have Laravel Herd running
- I also have laravel sail running
- I setup /etc/hosts for the .test domain to point to 127.0.0.1
Is my site being served from Laravel Herd's nginx or my docker nginx?


Thanks again for such a great product.

**Comment by mpociot** [comment]
@cs02lk1 

> I setup /etc/hosts for the .test domain to point to 127.0.0.1
> Is my site being served from Laravel Herd's nginx or my docker nginx?

By default, Herd intercepts all .test domains
